pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -424,7 +424,9 @@ func realScheduleMonitor(c *Cluster, mon *monConfig) (SchedulingResult, error) {
 
 	// setup affinity settings for pod scheduling
 	p := cephv1.GetMonPlacement(c.spec.Placement)
-	c.setPodPlacement(&d.Spec.Template.Spec, p, nil)
+	if !c.spec.Mon.AllowMultiplePerNode {
+		c.setPodPlacement(&d.Spec.Template.Spec, p, nil)
```

Comment:
Is this still needed? The issue may have already been resolved by #4457.

Reply:
i think its still needed because the placements still gets applied for the canary pods of mons.

Reply:
Without this change, I'm able to launch a cluster in minkube with 3 mons where `allowMultiplePerNode: true`. Can you explain more why you see this as necessary? I believe the mons set `preferredDuringScheduling` instead of `requiredDuringScheduling` for the anti-affinity when `allowMultiplePerNode: false`.

Reply:
yeah, I think we don't need this. Seems I overlooked the scheduling policy. Closing this PR.